contrib/runners/action_chain_runner/action_chain_runner.py

```diff
@@ -278,104 +278,210 @@ def pre_run(self):
         try:
             self.chain_holder.validate()
         except Exception as e:
-            raise runnerexceptions.ActionRunnerPreRunError(e.message)
+            raise runner_exc.ActionRunnerPreRunError(e.message)
 
     def run(self, action_parameters):
-        # holds final result we store.
+        # Run the action chain.
+        return self._run_chain(action_parameters)
+
+    def pause(self):
+        # Identify the list of action executions that are workflows and cascade pause.
+        for child_exec_id in self.execution.children:
+            child_exec = ActionExecution.get(id=child_exec_id, raise_exception=True)
+            if (child_exec.runner['name'] in action_constants.WORKFLOW_RUNNER_TYPES and
+                    child_exec.status == action_constants.LIVEACTION_STATUS_RUNNING):
+                action_service.request_pause(
+                    LiveAction.get(id=child_exec.liveaction['id']),
+                    self.context.get('user', None)
+                )
+
+        return (
+            action_constants.LIVEACTION_STATUS_PAUSING,
+            self.liveaction.result,
+            self.liveaction.context
+        )
+
+    def resume(self):
+        # Restore runner and action parameters since they are not provided on resume.
+        runner_parameters, action_parameters = param_utils.render_final_params(
+            self.runner_type_db.runner_parameters,
+            self.action.parameters,
+            self.liveaction.parameters,
+            self.liveaction.context
+        )
+
+        # Assign runner parameters needed for pre-run.
+        if runner_parameters:
+            self.runner_parameters = runner_parameters
+
+        # Restore chain holder if it is not initialized.
+        if not self.chain_holder:
+            self.pre_run()
+
+        # Change the status of the liveaction from resuming to running.
+        self.liveaction = action_service.update_status(
+            self.liveaction,
+            action_constants.LIVEACTION_STATUS_RUNNING,
+            publish=False
+        )
+
+        # Run the action chain.
+        return self._run_chain(action_parameters, resuming=True)
+
+    def _run_chain(self, action_parameters, resuming=False):
+        # Set chain status initially to success until chain cancels, fails, or times out below.
+        chain_status = action_constants.LIVEACTION_STATUS_SUCCEEDED
```

Comment:
I think setting it to failure by default would be safer and also act as "catch all" in case we accidentally fall through and don't explicitly set a status.